src/lib.rs

```diff
@@ -1,744 +1,345 @@
 
-impl<'lib> IRCompiler<'lib> {
-    pub fn new(
-        lib: &'lib libloading::Library,
-    ) -> Result<IRCompiler<'lib>, Box<dyn std::error::Error>> {
+    pub fn set_hitgroup_type(&mut self, hit_group_type: IRHitGroupType) {
         unsafe {
-            let funcs = IRCompilerFn {
-                alloc_compile_and_link: lib.get(b"IRCompilerAllocCompileAndLink")?,
-                create: lib.get(b"IRCompilerCreate")?,
-                destroy: lib.get(b"IRCompilerDestroy")?,
-                enable_geometry_and_tessellation_emulation: lib
-                    .get(b"IRCompilerEnableGeometryAndTessellationEmulation")?,
-                set_compatibility_flags: lib.get(b"IRCompilerSetCompatibilityFlags")?,
-                set_depth_feedback_configuration: lib
-                    .get(b"IRCompilerSetDepthFeedbackConfiguration")?,
-                set_dual_source_blending_configuration: lib
-                    .get(b"IRCompilerSetDualSourceBlendingConfiguration")?,
-                set_entry_point_name: lib.get(b"IRCompilerSetEntryPointName")?,
-                set_global_root_signature: lib.get(b"IRCompilerSetGlobalRootSignature")?,
-                set_has_geometry_shader: lib.get(b"IRCompilerSetHasGeometryShader")?,
-                set_hitgroup_arguments: lib.get(b"IRCompilerSetHitgroupArguments")?,
-                set_input_topology: lib.get(b"IRCompilerSetInputTopology")?,
-                set_local_root_signature: lib.get(b"IRCompilerSetLocalRootSignature")?,
-                set_minimum_deployment_target: lib.get(b"IRCompilerSetMinimumDeploymentTarget")?,
-                set_minimum_gpu_family: lib.get(b"IRCompilerSetMinimumGPUFamily")?,
-                set_shared_rt_arguments: lib.get(b"IRCompilerSetSharedRTArguments")?,
-                set_stage_in_generation_mode: lib.get(b"IRCompilerSetStageInGenerationMode")?,
-                set_stream_out_enabled: lib.get(b"IRCompilerSetStreamOutEnabled")?,
-                set_strip_cut_index: lib.get(b"IRCompilerSetStripCutIndex")?,
-                set_tessellation_enabled: lib.get(b"IRCompilerSetTessellationEnabled")?,
-                set_tessellator_output_primitive: lib
-                    .get(b"IRCompilerSetTessellatorOutputPrimitive")?,
-                set_validation_flags: lib.get(b"IRCompilerSetValidationFlags")?,
-                set_vertex_render_target_index_id: lib
-                    .get(b"IRCompilerSetVertexRenderTargetIndexID")?,
-                set_vertex_viewport_index_id: lib.get(b"IRCompilerSetVertexViewportIndexID")?,
-            };
+            self.funcs
+                .IRCompilerSetHitgroupType(self.me, hit_group_type)
+        }
+    }
 
-            let me = (funcs.create)();
+    pub fn synthesize_indirect_intersection_function(
+        &mut self,
+        target_metallib: &mut IRMetalLibBinary,
+    ) -> bool {
+        unsafe {
+            self.funcs
+                .IRMetalLibSynthesizeIndirectIntersectionFunction(self.me, target_metallib.me)
```

Comment:
Unsure how this is supposed to be called. But maybe this can become something more like:

```rust
let lib = self.create_lib_binary()?;
let success = IRMetalLibSynthesizeIndirectIntersectionFunction(lib.me);
if success {
    Ok(lib)
} else {
    Err(...)
}
```

Reply:
This looks like a new function in this PR so we should either:

1. Fix the issues with it directly;
2. Remove the function from this PR and introduce it in your followup PR.

Same for other code / review comments that are still open.

Reply:
Decided to remove the function, I'll implement it in a future pr